sender-stackdriver/src/test/java/zipkin2/reporter/stackdriver/ITStackdriverSender.java

```diff
@@ -13,90 +13,131 @@
  */
 package zipkin2.reporter.stackdriver;
 
-import com.google.devtools.cloudtrace.v2.BatchWriteSpansRequest;
-import com.google.devtools.cloudtrace.v2.TraceServiceGrpc;
-import com.google.protobuf.Empty;
-import io.grpc.stub.StreamObserver;
-import io.grpc.testing.GrpcServerRule;
-import java.util.concurrent.TimeUnit;
-import java.util.function.Consumer;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.devtools.cloudtrace.v1.GetTraceRequest;
+import com.google.devtools.cloudtrace.v1.Trace;
+import com.google.devtools.cloudtrace.v1.TraceServiceGrpc;
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.grpc.auth.MoreCallCredentials;
+import org.awaitility.Duration;
+import org.junit.After;
 import org.junit.Before;
-import org.junit.Rule;
 import org.junit.Test;
-import org.mockito.ArgumentCaptor;
-import org.mockito.stubbing.Answer;
+import zipkin2.CheckResult;
 import zipkin2.Span;
-import zipkin2.TestObjects;
 import zipkin2.reporter.AsyncReporter;
-import zipkin2.translation.stackdriver.SpanTranslator;
 
-import static java.util.Arrays.asList;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.mockito.Matchers.any;
-import static org.mockito.Mockito.doAnswer;
-import static org.mockito.Mockito.never;
-import static org.mockito.Mockito.spy;
-import static org.mockito.Mockito.verify;
+import static org.awaitility.Awaitility.await;
+import static zipkin2.TestObjects.FRONTEND;
+import static zipkin2.TestObjects.BACKEND;
+import static zipkin2.TestObjects.TODAY;
 
 /** Same as ITStackdriverSpanConsumer: tests everything wired together */
 public class ITStackdriverSender {
-  @Rule public final GrpcServerRule server = new GrpcServerRule().directExecutor();
-  TestTraceService traceService = spy(new TestTraceService());
-  String projectId = "test-project";
+  String projectId = "zipkin-gcp-ci";
+  GoogleCredentials credentials;
+  StackdriverSender sender;
+  StackdriverSender senderNoPermission;
   AsyncReporter<Span> reporter;
+  AsyncReporter<Span> reporterNoPermission;
+  TraceServiceGrpc.TraceServiceBlockingStub traceServiceGrpcV1;
+  Channel v1Channel;
 
   @Before
-  public void setUp() {
-    server.getServiceRegistry().addService(traceService);
+  public void setUp() throws IOException {
+  	// Application Default credential is configured using the GOOGLE_APPLICATION_CREDENTIALS env var
```

Comment:
Maybe ` or gcloud SDK when run locally.`

Reply:
I'd prefer to see an `assumeThat` here which will ignore the test vs break the build, when remote stuff is not around. That way you can craft a better error. PS I think the build is broke not sure intended or not.

Reply:
TIL :D updated to use `assumeThatCode` since `GoogleCredentials.getApplicationDefault` throws an exception that we need to catch.